libs/core/algorithms/include/hpx/parallel/util/loop.hpp

```diff
@@ -786,7 +821,7 @@ namespace hpx::parallel::util {
 
         // Helper class to repeatedly call a function a given number of times
         // starting from a given iterator position.
-        template <typename IterCat>
+        template <typename IterCat, typename Dummy = void>
```

Comment:
Same question as above: where is `loop_idx_n<IterCat, void>` used?

Reply:
I believe this was done so loop_pred can be templated as `loop_idx_n<Iter>` and `loop_idx_n<Iter, ExPolicy>`
Now that we have moved the dispatch to tag_invoke, I don't think this is required, I will look into replacing it with the older version.